src/hip/HIPStream.cpp

```diff
@@ -107,9 +107,9 @@ HIPStream<T>::~HIPStream()
 
 
 template <typename T>
-__global__ void init_kernel(T * a, T * b, T * c, T initA, T initB, T initC, int array_size)
+__global__ void init_kernel(T * a, T * b, T * c, T initA, T initB, T initC, intptr_t array_size)
```

Comment:
Change this to size_t to make it equal to CUDA for now. Need to test this on AMD GPUs.